pkg/components/httpproxy.go

```diff
@@ -32,6 +32,44 @@ func NewHTTPProxy(
 	spec ytv1.HTTPProxiesSpec,
 ) *HttpProxy {
 	l := cfgen.GetComponentLabeller(consts.HttpProxyType, spec.Role)
+	containerPorts := []corev1.ContainerPort{
+		{
+			Name:          consts.YTRPCPortName,
+			ContainerPort: consts.HTTPProxyRPCPort,
+			Protocol:      corev1.ProtocolTCP,
+		},
+		{
+			Name:          consts.HTTPPortName,
+			ContainerPort: ptr.Deref(spec.HttpPort, consts.HTTPProxyHTTPPort),
+			Protocol:      corev1.ProtocolTCP,
+		},
+		{
+			Name:          consts.HTTPSPortName,
+			ContainerPort: ptr.Deref(spec.HttpsPort, consts.HTTPProxyHTTPSPort),
+			Protocol:      corev1.ProtocolTCP,
+		},
+	}
+	var chytProxy *ytv1.CHYTProxySpec
+	if ytsaurus.GetClusterFeatures().HTTPProxyHaveChytAddress {
+		var err error
+		chytProxy, err = ytconfig.MakeChytProxySpec(spec)
+		if err != nil {
+			return nil
+		}
+	}
+
+	if chytProxy != nil {
+		containerPorts = append(containerPorts, corev1.ContainerPort{
+			Name:          consts.CHYTHttpProxyName,
+			ContainerPort: ptr.Deref(chytProxy.HttpPort, int32(consts.HTTPProxyChytHttpPort)),
```

Comment:
Maybe `ptr.Deref(...)`is not needed here? @koct9i

Reply:
Ah. That not chyt-proxy spec form spec but "spec" from config.

Reply:
I see no reason for ytconfig.MakeChytProxySpec
that could be written in place sorter and clearer

```
chytProxy := ptr.Deref(httpProxy.ChytProxy, ytv1.CHYTProxySpec{})
httpPort := ptr.Deref(chytProxy.HttpPort, consts.ChytDefaultHttpPort)
httspPort := ptr.Deref(chytProxy.HttpsPort, consts.ChytDefaultHttpsPort)
```

checking for zero port at this stage is redundant, such checks should be in validator

Reply:
And about zero, see comment: https://github.com/ytsaurus/ytsaurus-k8s-operator/pull/553#discussion_r2340654056, pt. 2